src/App/App.js

```diff
@@ -1,38 +1,58 @@
 import React, { Component } from 'react';
 import MovieList from '../MovieList/MovieList';
 import Header from '../Header/Header';
-import movieData from '../mockData';
 import './App.css';
 import MovieDetail from '../MovieDetail/MovieDetail';
+import { getAllMovies, getMovieDetails } from '../api-calls/api-calls';
 
 class App extends Component {
   constructor() {
     super();
     this.state = {
-      movies: movieData.movies,
-      selectedMovie: null
+      movies: [],
+      selectedMovie: null,
+      error:''
     }
   }
 
-  handleClick = (id) => {
-    const clickedMovie = this.state.movies.find(movie => movie.id === id);
-    this.setState({ selectedMovie: clickedMovie });
+  componentDidMount() {
+    getAllMovies()
+    .then(data => {
+      this.setState({ movies: data.movies })
+    })
+    .catch(() => this.setState(
+      { error: 'Something went wrong, please try again later.' }))
   }
 
   goBacktoMainView = () => {
     this.setState({ selectedMovie: null });
   }
 
+  handleClick = (id) => {
+    getMovieDetails(id)
+      .then(data => {
+        this.setState({ selectedMovie: data.movie })
+      })
+      .catch(() => this.setState(
```

Comment:
Same syntax note here - be consistent. Use the implicit return for both, or neither.